Framework/Algorithms/inc/MantidAlgorithms/SampleCorrections/MCInteractionVolume.h

```diff
@@ -46,8 +46,12 @@ class MANTID_ALGORITHMS_DLL MCInteractionVolume {
                              const Kernel::V3D &startPos,
                              const Kernel::V3D &endPos, double lambdaBefore,
                              double lambdaAfter) const;
+  std::string generateScatterPointStats() const;
+  Kernel::V3D generatePoint(Kernel::PseudoRandomNumberGenerator &rng) const;
 
 private:
+  mutable int m_sampleScatterPoints = 0;
```

Comment:
These shouldn't be marked `mutable` as they aren't `const` (and they are intended to be changed)

Reply:
Without the mutable keyword I get a compiler error:
'm_sampleScatterPoints' cannot be modified because it is being accessed through a const object

This is because the MCInteractionVolume object member inside the MCAbsorptionStrategy is defined as const:

class MANTID_ALGORITHMS_DLL MCAbsorptionStrategy {
public:
  MCAbsorptionStrategy(const IBeamProfile &beamProfile,
                       const API::Sample &sample, size_t nevents,
                       size_t maxScatterPtAttempts);
  std::tuple<double, double> calculate(Kernel::PseudoRandomNumberGenerator &rng,
                                       const Kernel::V3D &finalPos,
                                       double lambdaBefore, double lambdaAfter) const;
  std::string getDebugString() const { return debugString; };

private:
  const IBeamProfile &m_beamProfile;
  **const MCInteractionVolume m_scatterVol;**
  const size_t m_nevents;
  const size_t m_maxScatterAttempts;
  const double m_error;
  mutable std::string debugString;
};

Reply:
Hmmm, I missed that originally - so I genuinely had to go back and look up best practise for mutable [here](https://isocpp.org/wiki/faq/const-correctness#mutable-data-members)

In this case calling `generatePoint` has an observable change to the logical state (observable through `generateScatterPointStats`) 

That is to say a developer calling `generateScatterPointStats` on a const object will have the object change underneath them which can be unexpected. So someone might naively assume

```
const MCInteractionVolume calc;
calc.generateScatterPointStats()
calc.generateScatterPointStats() // These are equal
calc.generatePoint() 
calc.generateScatterPointStats // Now differ despite the object being const
```

Therefore I think we should drop the const-ness on `generatePoint` and remove the `const` qualifier in MCAbsorptionStratergy. Especially since this is a class whose sole purpose is for calculating (so shouldn't be a proper-gating API change)

I appreciate this is one of the more nit-picky ones, but `debugString` is another observable which is mutated too (and I'm cautious of the number of things we do to simply avoid dropping const)

Reply:
OK I've done that. There was a small further knock on effect - I also had to drop the const qualifiers on MCInteractionVolume::calculateAbsorption() and MCAbsorptionStrategy::calculate()